### PR TITLE
PPC Refactor: Utilize Bolt_Boltpay_Model_BoltOrder and Bolt_Boltpay_Block_Checkout_Boltpay

### PR DESCRIPTION
**Pull request by mccottry**
# Description
PPC currently exist in its own island doing a lot of repeat code.  This is not desirable.  Instead of repeating code, we send PPC Bolt Order creation down the same path as all other order creation.

Fixes: https://app.asana.com/0/941895179897714/1161793187073237

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Test
- [x] Refactor

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php [hunk 1]
```diff
@@ -207,4 +216,23 @@ public function getProductMaxQty()
...
+     *
+     * @return string
+     */
+    public function getProductJSON()
+    {
+        /** @var Mage_Catalog_Model_Product $product */
+        $product = Mage::registry('current_product');
+        return Mage::helper('core')->jsonEncode(
+            array(
+                'id'         => $product->getId(),
```
Comment (adentong) [inline app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php]: Let's align the arrows.

File: app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php [hunk 2]
```diff
@@ -26,31 +26,32 @@
...
+     * @return string JSON encoded Bolt cart data containing current product
+     * @deprecated
      */
     public function getCartDataJsForProductPage()
     {
         try {
             $currency = Mage::app()->getStore()->getCurrentCurrencyCode();
 
             $productCheckoutCartItem = [];
-            $totalAmount = (float) 0;
```
Comment (adentong) [inline app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php]: Was the removal of this line intentional?

File: app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php (hunk 1 above)
Comment (mccottry) [reply app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php]: no problem.

File: app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php (hunk 2 above)
Comment (mccottry) [reply app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php]: @adentong 
Yes.  The initialization of this occurs just below here:
https://github.com/BoltApp/bolt-magento1/blob/a77a5febaecd03ec2973513acd591849fd50a557/app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php#L75

This line is a useless (unused) piece of code.

File: app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php (hunk 1 above)
Comment (mccottry) [reply app/code/community/Bolt/Boltpay/Block/Catalog/Product/Boltpay.php]: @adentong Arrows have been aligned.

**Pull request merged**